### PR TITLE
Rails 5.1 compatibility

### PR DESCRIPTION
**Pull request by samphippen**

### CODE REVIEW

**Comment by samphippen** [comment]
@yui-knk btw, here's the current failure output when I run `RAILS_VERSION=5.1.0.beta1 ./script/run_build` in my clone of RSpec Rails: https://gist.github.com/samphippen/bcc02764dca82bb8f416a798f9561d8e

**Comment by myronmarston** [comment]
@samphippen any reason not to update this to 5.1.0.rc1?  That's going to be closer to the 5.1 final release, and it's possible the build failures are related to a regression in beta1 that's fixed in rc1.

**Comment by samphippen** [comment]
No reason, just haven't had the time.

**Comment by yui-knk** [comment]
I create https://github.com/rspec/rspec-rails/pull/1797. Can you merge this?

**Comment by samphippen** [comment]
@rspec/rspec this is now ready for review. I've tested it locally against 5.1 and 5.0 and everything still works. Travis is cooking the other versions, but I don't expect much variance there.

File: features/request_specs/request_spec.feature [hunk 1]
```diff
@@ -55,7 +56,38 @@ Feature: request spec
...
+        it "does not render a different template" do
+          get "/widgets/new"
+          expect(response).to_not render_template(:show)
+        end
+      end
+      """
+    When I run `rspec spec/requests/widget_management_spec.rb`
+    Then the example should pass
+
+
```
Comment (JonRowe) [inline features/request_specs/request_spec.feature]: Extra ws here

File: features/matchers/have_http_status_matcher.feature [hunk 2]
```diff
@@ -136,7 +162,6 @@ Feature: `have_http_status` matcher
           visit "/widgets/new"
           expect(page).to have_http_status(200)
 
-          fill_in "Name", :with => "My Widget"
```
Comment (JonRowe) [inline features/matchers/have_http_status_matcher.feature]: unrelated?

File: lib/generators/rspec/scaffold/templates/edit_spec.rb [hunk 3]
```diff
@@ -16,8 +16,12 @@
     assert_select "form[action=?][method=?]", <%= ns_file_name %>_path(@<%= ns_file_name %>), "post" do
 <% for attribute in output_attributes -%>
       <%- name = attribute.respond_to?(:column_name) ? attribute.column_name : attribute.name %>
+<% if Rails.version.to_f >= 5.1 -%>
```
Comment (JonRowe) [inline lib/generators/rspec/scaffold/templates/edit_spec.rb]: Worth indenting to match above? Makes it slightly clearer too.

File: lib/generators/rspec/scaffold/templates/new_spec.rb [hunk 4]
```diff
@@ -15,8 +15,12 @@
     assert_select "form[action=?][method=?]", <%= index_helper %>_path, "post" do
 <% for attribute in output_attributes -%>
       <%- name = attribute.respond_to?(:column_name) ? attribute.column_name : attribute.name %>
+<% if Rails.version.to_f >= 5.1 -%>
```
Comment (JonRowe) [inline lib/generators/rspec/scaffold/templates/new_spec.rb]: As above

**Comment by JonRowe** [comment]
Seems sane to me but I've not run it, if you're happy with it I'm happy with it. I've left a few non blocking comments on style.

File: Rakefile [hunk 5]
```diff
@@ -11,6 +11,15 @@ require 'rspec'
 require 'rspec/core/rake_task'
 require 'cucumber/rake/task'
 
+def rails_template_command
+  require "rails"
+  if Rails.version.to_f >= 5.1
```
Comment (yui-knk) [inline Rakefile]: I think `if Rails.version.to_f >= 5.0` is better, because ` "rails:template"` was deprecated on Rails 5.0.

File: features/matchers/have_http_status_matcher.feature (hunk 2 above)
Comment (samphippen) [inline features/matchers/have_http_status_matcher.feature]: has no effect and breaks in Rails 5.1 for really weird reasons.

File: lib/generators/rspec/scaffold/templates/edit_spec.rb (hunk 3 above)
Comment (samphippen) [inline lib/generators/rspec/scaffold/templates/edit_spec.rb]: The problem here is that indentation here affects the output. We want the output to look 'normal' do we don't indent here.

File: lib/generators/rspec/scaffold/templates/new_spec.rb (hunk 4 above)
Comment (samphippen) [inline lib/generators/rspec/scaffold/templates/new_spec.rb]: same

File: Rakefile (hunk 5 above)
Comment (samphippen) [inline Rakefile]: if it passes on CI it's good with me :)

**Comment by samphippen** [comment]
@JonRowe I've just pushed some review tweaks and answered a few of your review pieces. WDYT?

File: features/matchers/have_http_status_matcher.feature (hunk 2 above)
Comment (JonRowe) [inline features/matchers/have_http_status_matcher.feature]: Is that indicative that we ave an issue here?

File: lib/generators/rspec/scaffold/templates/edit_spec.rb (hunk 3 above)
Comment (JonRowe) [inline lib/generators/rspec/scaffold/templates/edit_spec.rb]: You could indent inside the tag if thats your concern?

File: features/matchers/have_http_status_matcher.feature (hunk 2 above)
Comment (samphippen) [inline features/matchers/have_http_status_matcher.feature]: There's no issue here. In rails 5.1 they've changed the form generation logic so this finder fails. However, this `fill_in` has nothing to do with the test. I think we should remove it 

1. for compatibility
2. because it doesn't aid in being documenting of this feature

File: lib/generators/rspec/scaffold/templates/edit_spec.rb (hunk 3 above)
Comment (samphippen) [inline lib/generators/rspec/scaffold/templates/edit_spec.rb]: Can you show how you'd like this to look, I'm not sure I fully follow?

**Comment by samphippen** [comment]
@JonRowe I'm gonna go ahead and merge this. If you feel particularly strongly on the indentation or the deleted line, I think we can open a fresh PR.
